x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/TransportGetShutdownStatusAction.java

```diff
@@ -66,13 +68,36 @@ protected void masterOperation(
         if (nodesShutdownMetadata == null) {
             response = new GetShutdownStatusAction.Response(new ArrayList<>());
         } else if (request.getNodeIds().length == 0) {
-            response = new GetShutdownStatusAction.Response(new ArrayList<>(nodesShutdownMetadata.getAllNodeMetadataMap().values()));
+            final List<SingleNodeShutdownStatus> shutdownStatuses = new ArrayList<>(nodesShutdownMetadata.getAllNodeMetadataMap().size());
+            nodesShutdownMetadata.getAllNodeMetadataMap()
+                .values()
+                .forEach(
+                    ns -> shutdownStatuses.add(
+                        new SingleNodeShutdownStatus(
+                            ns,
+                            new ShutdownShardMigrationStatus(),
+                            new ShutdownPersistentTasksStatus(),
+                            new ShutdownPluginsStatus()
+                        )
+                    )
+                );
+            response = new GetShutdownStatusAction.Response(shutdownStatuses);
         } else {
-            Map<String, SingleNodeShutdownMetadata> nodeShutdownMetadataMap = nodesShutdownMetadata.getAllNodeMetadataMap();
-            final List<SingleNodeShutdownMetadata> shutdownStatuses = Arrays.stream(request.getNodeIds())
+            final List<SingleNodeShutdownStatus> shutdownStatuses = new ArrayList<>();
+            final Map<String, SingleNodeShutdownMetadata> nodeShutdownMetadataMap = nodesShutdownMetadata.getAllNodeMetadataMap();
+            Arrays.stream(request.getNodeIds())
                 .map(nodeShutdownMetadataMap::get)
                 .filter(Objects::nonNull)
-                .collect(Collectors.toList());
+                .forEach(
+                    ns -> shutdownStatuses.add(
+                        new SingleNodeShutdownStatus(
+                            ns,
+                            new ShutdownShardMigrationStatus(),
+                            new ShutdownPersistentTasksStatus(),
+                            new ShutdownPluginsStatus()
+                        )
+                    )
+                );
```

Comment:
What does replacing the `.collect(Collectors.toList())` with `.forEach(/* add the item to an existing list */)` buy us?

Reply:
absolutely nothing, I can change this back to a `.map` and then `.collect` if you like, it was just done quickly :)

Reply:
Haha okay! I think I would prefer that (and above in the same file). I know some folks who disagree but I'm of the opinion that map/collect is easier to reason about than a side-effecting foreach.

Reply:
Yep, I agree with you, I'll change this and then merge this tomorrow